lemur/common/celery.py

```diff
@@ -316,7 +315,7 @@ def fetch_acme_cert(id, notify_reissue_cert=None):
                 send_pending_failure_notification(
                     pending_cert, notify_owner=pending_cert.notify
                 )
-                if notify_reissue_cert is not None:
+                if pending_cert.notify and pending_cert.replaces is not None:
```

Comment:
Does anyone know if `pending_cert.notify` will be set at this point? I haven't dug into the code to check yet.

Reply:
my understanding is that it would follow what was provided in the UI or API request.

Reply:
I'm interested specifically in the case of reissued certs - at what point in time is the `notify` field of the new cert copied over from the prior cert? I'm reasonable sure it's already set at this point (since line 316 uses `.notify` as well, which I didn't initially notice), but I also didn't yet identify the code that takes care of setting it.

Reply:
Thanks for tagging me @jtschladen! Ugh, sorry it errored, admittedly I hadn't tested passing the full cert object -- I'll be more careful in the future. I believe the cert copy happens as follows

Reissue task dumps old cert's data (I believe this includes the `notify` field):

https://github.com/Netflix/lemur/blob/31774a4f03383b9afa69494c6d63830c84663c95/lemur/certificates/service.py#L866

Then after adjustments to fields that aren't `notify`, it reaches the creation call:

https://github.com/Netflix/lemur/blob/31774a4f03383b9afa69494c6d63830c84663c95/lemur/certificates/service.py#L935

Then the created cert is committed to Postgres (either as `PendingCertificate` or full `Certificate`):

https://github.com/Netflix/lemur/blob/31774a4f03383b9afa69494c6d63830c84663c95/lemur/certificates/service.py#L489

All that should be happening before the [acme fetch task runs](https://github.com/Netflix/lemur/blob/31774a4f03383b9afa69494c6d63830c84663c95/lemur/certificates/service.py#L518).

Hope that seems to be correct! Thanks for catching this.

Reply:
Thanks - that sounds right to me. Appreciate the input!